app/org/sagebionetworks/bridge/services/ScheduledActivityService.java

```diff
@@ -228,11 +228,13 @@ protected DateTime getDateTime() {
             ScheduledActivity activity = scheduledActivities.get(i);
             ScheduledActivity dbActivity = dbMap.remove(activity.getGuid());
             
-            if (dbActivity != null && !UPDATABLE_STATUSES.contains(dbActivity.getStatus())) {
-                // Once the activity is in the database and is in a non-updatable state, we should use the one from the
-                // database. Otherwise, either (a) it doesn't exist yet and needs to be persisted or (b) the user
-                // hasn't interacted with it yet, so we can safely replace it with the newly generated one, which may
-                // have updated schemas or surveys.
+            boolean updatableStatus = dbActivity != null && UPDATABLE_STATUSES.contains(dbActivity.getStatus());
+            
+            if (dbActivity != null && (!updatableStatus || dbActivity.getClientData() != null)) {
```

Comment:
Minor: The dbActivity != null clause in updatableStatus is redundant with the dbActivity != null in the if statement. You could simplify the whole thing to

if (dbActivity != null && (!UPDATABLE_STATUSES.contains(dbActivity.getStatus()) || dbActivity.getClientData() != null)) {

Reply:
Changed